lms/djangoapps/instructor/permissions.py

```diff
@@ -82,3 +86,32 @@ def has_permission(self, request, view):
         course = get_course_by_id(CourseKey.from_string(view.kwargs.get('course_id')))
         permission = getattr(view, 'permission_name', None)
         return request.user.has_perm(permission, course)
+
+
+class ForumAdminRequiresInstructorAccess(BasePermission):
+    """
+    default roles require either (staff & forum admin) or (instructor)
+    User should be forum-admin and staff to access this endpoint.
+
+    But if request rolename is  FORUM_ROLE_ADMINISTRATOR, then user must also have
+    instructor-level access to proceed.
+    """
+    def has_permission(self, request, view):
+        rolename = request.data.get('rolename')
+        course_id = view.kwargs.get('course_id')
+        course = get_course_by_id(CourseKey.from_string(course_id))
+
+        has_instructor_access = has_access(request.user, 'instructor', course)
+        has_forum_admin = has_forum_access(
+            request.user, course_id, FORUM_ROLE_ADMINISTRATOR
+        )
+
+        # default roles require either (staff & forum admin) or (instructor)
+        if not (has_forum_admin or has_instructor_access):
+            raise PermissionDenied("Operation requires staff & forum admin or instructor access")
+
+        # EXCEPT FORUM_ROLE_ADMINISTRATOR requires (instructor)
+        if rolename == FORUM_ROLE_ADMINISTRATOR and not has_instructor_access:
+            raise PermissionDenied("Operation requires instructor access.")
+
+        return True
```

Comment:
Instead of defaulting to `True` I think it's better to default to false and have a check for each of the combinations we're okay with.  I'm having a little bit of trouble understanding how the two checks are different.  If you're already checking to see if the user is a `FORUM_ROLE_ADMINSTATOR` do you care about whether or not that parameter is passed into the request?